shoppingcart/Dockerfile

```diff
@@ -1,8 +1,10 @@
 FROM python:3.8.0-slim
+
 COPY . /app
 RUN apt-get update \
-&& apt-get install gcc -y \
-&& apt-get clean
+    && apt-get install gcc -y \
+    && apt-get clean
+
 WORKDIR app
 RUN pip install --user -r requirements.txt
```

Comment:
gcc is needed for pip install here or also to run the shopping cart? The docker image is about 360MB in size.
I started to have multi stage builds if I can, and if I use a multi stage build here this is down to 190MB. For sure, this only works if gcc is not needed to run the shopping cart. wdyt @sleipnir ?

```
python-support-tck-multi   latest   ea55b48a53d8    2 seconds ago       193MB
python-support-tck         latest   c11064815b7e    2 minutes ago       364MB
```

multistage
```
FROM python:3.8-slim
COPY . /app
RUN apt-get update \
&& apt-get install gcc -y \
&& apt-get clean
WORKDIR app
RUN pip install --user -r requirements.txt
# multi stage build
FROM python:3.8-slim
ENTRYPOINT python shopping_cart.py
```

Reply:
Here I didn't expect it to be really definitive, I just want to play with it and then better. I see no problems with the multi stage approach and it was already in my all future.